frontend/packages/ux-editor/src/hooks/useLayoutSetIcon.tsx

```diff
@@ -9,17 +9,37 @@ import {
 } from '@studio/icons';
 import type { LayoutSetModel } from 'app-shared/types/api/dto/LayoutSetModel';
 import type { StudioIconCardIconColors } from '@studio/components-legacy/src/components/StudioIconCard/StudioIconCard';
+import type { BpmnTaskType } from 'app-shared/types/BpmnTaskType';
 
-export const useLayoutSetIcon = (
-  layoutSetModel: LayoutSetModel,
-): { icon: ReactElement; iconColor: StudioIconCardIconColors } => {
-  if (layoutSetModel.type == 'subform') return { icon: <ClipboardIcon />, iconColor: 'blue' };
+type IconMetaData = {
+  icon: ReactElement;
+  iconColor: StudioIconCardIconColors;
+};
+
+export const useLayoutSetIcon = (layoutSetModel: LayoutSetModel): IconMetaData => {
+  return (
+    iconByLayoutSetModelType[layoutSetModel.type] ||
+    iconByTaskIdMap[layoutSetModel.task?.id] ||
+    iconByTaskTypeMap[layoutSetModel.task?.type] ||
+    defaultIcon
+  );
+};
```

Comment:
```suggestion
export const useLayoutSetIcon = (layoutSetModel: LayoutSetModel): IconMetaData => {
  const { type, task } = layoutSetModel; 
  return (
    iconByLayoutSetModelType[type] ||
    iconByTaskIdMap[task?.id] ||
    iconByTaskTypeMap[task?.type] ||
    defaultIcon
  );
};
```
What do you think about destructuring it?